plugins/genesys/fu-genesys-scaler-device.c

```diff
@@ -0,0 +1,1806 @@
+						     self->public_key,
+						     sizeof(self->public_key),
+						     error))
+		return FALSE;
+	guid =
+	    fwupd_guid_hash_data(self->public_key, sizeof(self->public_key), FWUPD_GUID_FLAG_NONE);
+
+	if (!fu_genesys_scaler_device_get_version(self, buf, sizeof(buf), error))
+		return FALSE;
+	version =
+	    fu_common_strsafe((const gchar *)&buf[1], 6); /* ?RIM123; where ? is 0x06 (length?) */
+
+	fu_device_set_version(device, version);
+	fu_device_set_version_format(device, FWUPD_VERSION_FORMAT_PLAIN);
+	fu_device_set_logical_id(device, "scaler");
+	guid_upper = g_ascii_strup(guid, -1);
+	instance_id = g_strdup_printf("GENESYS_CPU_MODEL\\MCPU_TSUM_G&PUBKEY_%s", guid_upper);
+	fu_device_add_instance_id(device, instance_id);
+	fu_device_add_flag(device, FWUPD_DEVICE_FLAG_UPDATABLE);
+
+	self->vc.req_read = GENESYS_SCALER_MSTAR_READ;
+	self->vc.req_write = GENESYS_SCALER_MSTAR_WRITE;
+	if (self->level != 1) {
+		self->vc.req_read += 3;
+		self->vc.req_write += 3;
+	}
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_genesys_scaler_device_open(FuDevice *device, GError **error)
+{
+	FuDevice *parent_device = fu_device_get_parent(device);
+
+	return fu_device_open(parent_device, error);
+}
+
+static gboolean
+fu_genesys_scaler_device_close(FuDevice *device, GError **error)
+{
+	FuDevice *parent_device = fu_device_get_parent(device);
+
+	return fu_device_open(parent_device, error);
+}
+
+static gboolean
+fu_genesys_scaler_device_setup(FuDevice *device, GError **error)
+{
+	FuGenesysScalerDevice *self = FU_GENESYS_SCALER_DEVICE(device);
+	guint32 sector_size;
+	guint32 page_size;
+
+	self->cfi_device = fu_cfi_device_new(fu_device_get_context(FU_DEVICE(self)), "C84016");
+	if (!fu_device_probe(FU_DEVICE(self->cfi_device), error))
+		return FALSE;
+
+	sector_size = fu_cfi_device_get_sector_size(self->cfi_device);
+	if (sector_size != 0)
+		self->sector_size = sector_size;
+	page_size = fu_cfi_device_get_page_size(self->cfi_device);
+	if (page_size != 0)
+		self->page_size = page_size;
+
+	if (fu_device_get_firmware_size_max(device) == 0) {
+		guint64 size_max = fu_device_get_firmware_size_max(FU_DEVICE(self->cfi_device));
+
+		if (size_max == 0)
+			size_max = 0x400000;
+
+		if (fu_device_has_flag(device, FWUPD_DEVICE_FLAG_DUAL_IMAGE))
+			size_max /= 2;
+
+		fu_device_set_firmware_size_max(FU_DEVICE(self), size_max);
+	}
+
+	/* success */
+	return TRUE;
+}
+
+static GBytes *
+fu_genesys_scaler_device_dump_firmware(FuDevice *device, FuProgress *progress, GError **error)
+{
+	FuGenesysScalerDevice *self = FU_GENESYS_SCALER_DEVICE(device);
+	gsize size = fu_device_get_firmware_size_max(device);
+	guint addr = 0x000000;
+	g_autofree guint8 *buf = NULL;
+	g_autoptr(FuDeviceLocker) locker = NULL;
+
+	if (fu_device_has_flag(device, FWUPD_DEVICE_FLAG_DUAL_IMAGE))
+		addr = 0x200000;
+
+	/* progress */
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_add_flag(progress, FU_PROGRESS_FLAG_GUESSED);
```

Comment:
can you try this and get the daemon-provided values? then you can remove `GUESSED`

Reply:
Does the daemon not give you the values required, e.g. `raw timing data was { 0.002, 21.413, 0.595 } -- steps were set as [ 10 80 10 ] but should have been [ 1 97 2 ] at ../plugins/foo/fu-foo-device.c:3172`

Reply:
I am giving a try. does it need the --verbose flag?

Reply:
It might if you're running with -Dsupported=true